### PR TITLE
[Core] Remove Dimension

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
As agreed before, this will finally remove the `Dimension` method from the geometry classes as well as the `GeometryData` and `GeometryDimension` ones.

Related to #10445.

### CODE REVIEW

**Comment by loumalouomega** [comment]
At last

**Review by loumalouomega — DISMISSED** [review]
I assume it will break something in the CI, anyway, approving

File: CMakeLists.txt [hunk 1]
```diff
@@ -597,7 +597,7 @@ endforeach()
 message("\n***********************************************************************\n")
 
 message("\nList of upcoming API-breaking and behavior changes:")
-message("\t Geometry 'Dimension' deprecation: https://github.com/KratosMultiphysics/Kratos/pull/10445")
+message("\t Geometry 'Dimension' deprecation: https://github.com/KratosMultiphysics/Kratos/pull/10977")
```
Comment (loumalouomega) [inline CMakeLists.txt]:
```suggestion
message("\t Geometry 'Dimension' removed: https://github.com/KratosMultiphysics/Kratos/pull/10977")
```

**Comment by matekelemen** [comment]
Does this mean I'll be able to see my compiler output again? You're a life saver! <3

**Review by roigcarlo — APPROVED** [review]

**Comment by loumalouomega** [comment]
Yay

**Pull request merged**